cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/sql/jdbc/DataSourceInstantiator.java

```diff
@@ -38,24 +40,26 @@
 import java.sql.SQLException;
 import java.util.Map;
 import java.util.Properties;
-import java.util.function.Supplier;
 import javax.sql.DataSource;
 
 /**
  * Class to instantiate the {@link DataSource} for the sql related structured table.
  */
-public class DataSourceInstantiator implements Supplier<DataSource> {
+public class DataSourceInstantiator implements Provider<DataSource> {
```

Comment:
Since it implements guice `Provider`, you should name the class `DataSourceProvider`.